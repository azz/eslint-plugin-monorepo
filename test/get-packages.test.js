import getPackages from '../src/get-packages';
import path from 'path';

const LERNA_DIR = path.join(__dirname, 'fixture/lerna');
const YARN_DIR = path.join(__dirname, 'fixture/yarn');

expect.addSnapshotSerializer({
  test: value =>
    typeof value === 'string' &&
    (value.indexOf('\\') > -1 || value.indexOf(process.cwd()) > -1),
  print: (value, serializer) =>
    serializer(value.replace(process.cwd(), '<cwd>').replace(/\\/g, '/')),
});

describe('getPackages()', () => {
  test('lerna matches snapshot', () => {
    expect(getPackages(LERNA_DIR)).toMatchSnapshot();
  });

  test('yarn matches snapshot', () => {
    expect(getPackages(YARN_DIR)).toMatchSnapshot();
  });

  test('returns empty array when no packages', () => {
    expect(getPackages(__dirname)).toEqual([]);
  });
});
