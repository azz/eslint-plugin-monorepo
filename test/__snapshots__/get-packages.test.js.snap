// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPackages() lerna matches snapshot 1`] = `
Array [
  Object {
    "fsPath": "<cwd>/test/fixture/lerna/packages/bar",
    "name": "bar",
  },
  Object {
    "fsPath": "<cwd>/test/fixture/lerna/packages/foo",
    "name": "foo",
  },
]
`;

exports[`getPackages() yarn matches snapshot 1`] = `
Array [
  Object {
    "fsPath": "<cwd>/test/fixture/yarn/packages/bar",
    "name": "bar",
  },
  Object {
    "fsPath": "<cwd>/test/fixture/yarn/packages/foo",
    "name": "foo",
  },
]
`;
